components-registry-service-server/src/main/kotlin/org/octopusden/octopus/components/registry/server/service/BuildSystemMetricsService.kt

```diff
@@ -0,0 +1,41 @@
+package org.octopusden.octopus.components.registry.server.service
+
+import io.micrometer.core.instrument.Gauge
+import io.micrometer.core.instrument.MeterRegistry
+import org.octopusden.octopus.components.registry.core.dto.BuildSystem
+import org.springframework.stereotype.Service
+import java.util.concurrent.ConcurrentHashMap
+
+/**
+ * Service for collecting metrics about components count by build system
+ */
+@Service
+class BuildSystemMetricsService(
```

Comment:
Why do we create separate service for updateMetrics method?

I suppose it's better to do it "in place" (somewhere in EscrowConfigurationLoader,  parseAndValidateConfiguration or loadComponentsFromConfigObject for instance).

Also in org.octopusden.octopus.components.registry.server.service there are interfaces of services, not implementations.

Reply:
@CamaradeRoman Should the registration of metrics also be done in EscrowConfigurationLoader?

Reply:
@ozonophore , many libraries delivered interface which is implemented directly on business service instead of separate one. Ones of these - HealthIndicator, UserService in Spring. If you want to pick it in one place, you may create you own Metics interface and inject it to you own handler

Reply:
@pgorbachev

Reply:
> @CamaradeRoman Should the registration of metrics also be done in EscrowConfigurationLoader?

@ozonophore, not in EscrowConfigurationLoader as it is a part of core module, but in ComponentRegistryResolver in server module. I have edited my comment.

Registration of metric could be done in ComponentRegistryResolver.